client/asset/eth/multirpc.go

```diff
@@ -569,12 +569,82 @@ func (m *multiRPCClient) voidUnusedNonce() {
 	m.lastNonce.stamp = time.Time{}
 }
 
-func (m *multiRPCClient) reconfigure(ctx context.Context, settings map[string]string) error {
+// createAndCheckProviders creates and connects to providers. It checks that
+// unknown providers have a sufficient api to trade and saves good providers to
+// file. One bad provider or connect problem will cause this to error.
+func createAndCheckProviders(walletDir string, endpoints []string, net dex.Network,
+	log dex.Logger, ctx context.Context) error {
+	var localCP map[string]bool
+	path := filepath.Join(walletDir, "compliant-providers.json")
+	b, err := os.ReadFile(path)
+	if err == nil {
+		if err := json.Unmarshal(b, &localCP); err != nil {
+			log.Warnf("Couldn't parse compliant providers file: %v", err)
+		}
+	}
+	if localCP == nil {
+		localCP = make(map[string]bool)
+	}
+
+	var writeLocalCP bool
+
+	var unknownEndpoints []string
+	for _, p := range endpoints {
+		d := domain(p)
+		if localCP[d] {
+			continue
+		}
+		writeLocalCP = true
+		localCP[d] = true
+		if _, known := compliantProviders[d]; known {
+			continue
+		}
+		unknownEndpoints = append(unknownEndpoints, p)
+	}
+
+	providers := []*provider{}
```

Comment:
I think this slice should just be declared by the connectProviders call.

Reply:
Oh yeah, I was thinking about returning this and not disconnecting sometimes, but it was messy so I didn't do that. Removing this unneeded declaration.